src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts

```diff
@@ -27,6 +27,16 @@ class ModalCompoundHetStore {
   }
 
   public setCurrentMode(modeType: ModeTypes) {
+    if (!modeType) {
+      this.currentMode = undefined
+    }
+
+    if (this.currentMode === modeType) {
```

Comment:
something is wrong here. `this.resetCurrentMode()` does the same thing that you do on 31 line. So if you set mode to undefined, you don't need to call `resetCurrentMode`. Maybe `if (! modeType || this.currentMode === modeType)` ?